src/mcts/search.cc

```diff
@@ -216,16 +224,29 @@ void Search::SendMovesStats() const {
         << ") ";
 
     oss << "(Q+U: " << std::setw(8) << std::setprecision(5)
-        << node->GetQ(parent_q) +
-               node->GetU() * kCpuct *
-                   std::sqrt(
-                       std::max(node->GetParent()->GetChildrenVisits(), 1u))
-        << ") ";
+        << node->GetQ(parent_q) + puct_u_coeff * node->GetU() << ") ";
     info.comment = oss.str();
     info_callback_(info);
   }
 }
 
+std::string Search::GetCachedFirstPlyValue(const Node* node) const {
```

Comment:
Function "Get node from cache, but only if it's first ply move, extract V, and convert to string, as percentage with precision of 2" does a bit too much (inspired by [this](http://php.net/manual/en/function.gzgetss.php)? :) ).

What I suggest to do:
1. Make it return NNCacheLock(), so that caller could extract either V or P if needed. (you'll need `return std::move(nneval)`).
    If you object doing that, return at least numeric value of V (as `optional<float>`).
2. Keep all string formatting in SendMovesStats().
3. It seems that it would be useful also to support cases other than first ply depth, and it's doable.
   (whole path from node to root_node_ can be restored, then applied to history).
   It's not necessary to implement that (but go ahead if you want) but I think "FirstPly" part should go from the function name. If you won't implement that, keep the assert and maybe add a short comment that it can be implemented by restoring path from root node to node if needed.
(that item is also debatable)

4. With all those changes, function should be called something like `GetCachedNodeRequestForDebug`, or something less clumsy. The `ForDebug` part (also debatable) is to hint people that it's not optimized and slow.

Reply:
I guess I broadly fall in the category of "[we aren't gonna need it](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)" and "do the simplest thing that could possibly work" (DTSTTCPW).

Although it may be useful in the future, I see no reason to implement things we don't even have a usecase for yet. I'm fine with designing the *current* use to be forward compatible (easy to modify, when needed), but I don't see the sense in actually implementing otherwise unused code. I guess I'm saying that "FirstPly" should stay in the function name, since we don't actually use it otherwise.

So I guess I'll go the `return optional<float>` route for now (not the slightest bit surprised that the string return value wasn't approved (:    ), and if we later need to generalize it, then sure we can do that relatively easily.